syntax = "proto2";

message transaction_t {
	required fixed32 sender_id = 1;
	required fixed32 receiver_id = 2;
    required float amount = 3;
}
		
message	message_t {
	required fixed32 client_id = 1;
	required timestamp_t timestamp = 2;
	required transaction_t transaction = 3;
}

message timestamp_t {
	required fixed64 seconds = 1;
	required fixed32 nanos = 2;
}

message request_t {
	required fixed32 type = 1;
}
